shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -727,6 +872,42 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     }
   }
 
+  private def getSupportedSchemaFromUnsupported(
+     cachedAttributes: Seq[Attribute],
+     requestedAttributes: Seq[Attribute] = Seq.empty): (Seq[Attribute], Seq[Attribute]) = {
+    // we only handle CalendarIntervalType and NullType ATM
+    // convert it to a supported type
+    val mapping = new mutable.HashMap[ExprId, List[Attribute]]()
+    val newCachedAttributes = cachedAttributes.flatMap {
+      attribute =>
+        if (attribute.dataType == DataTypes.CalendarIntervalType) {
+          val list = List(AttributeReference(attribute.name + "_cit_months",
+            DataTypes.IntegerType, attribute.nullable,
+            metadata = attribute.metadata)().asInstanceOf[Attribute],
+            AttributeReference(attribute.name + "_cit_days",
+              DataTypes.IntegerType, attribute.nullable,
+              metadata = attribute.metadata)().asInstanceOf[Attribute],
+            AttributeReference(attribute.name + "_cit_ms",
+              DataTypes.LongType, attribute.nullable,
+              metadata = attribute.metadata)().asInstanceOf[Attribute])
+          mapping.put(attribute.exprId, list)
+          list
+        } else if (attribute.dataType == DataTypes.NullType) {
+          val list = List(AttributeReference(attribute.name + "_nulltype", DataTypes.IntegerType,
+            attribute.nullable, metadata = attribute.metadata)().asInstanceOf[Attribute])
+          mapping.put(attribute.exprId, list)
+          list
+        }
+        else {
```

Comment:
Here too it looks like it does not support nested types with unsupported types in them

Reply:
Can we do that as a part of a follow-on?